sled-agent/src/instance.rs

```diff
@@ -176,6 +176,9 @@ struct InstanceInner {
     // Properties visible to Propolis
     properties: propolis_client::api::InstanceProperties,
 
+    // The ID of the Propolis server (and zone) running this instance
+    propolis_id: Uuid,
```

Comment:
So, to clarify, we have:

- `InstanceInner.id` - this is the UUID of the Instance (stable across migrations)
- `InstanceInner.properties.id` - this is the same as the top-level ID
- and now: `InstanceInner.propolis_id` - this is the UUID of propolis (NOT stable across Migrations)

If `InstanceInner.id` and `InstanceInner.properties.id` are both the "stable" Instance UUID, should we remove `InstanceInner.id`?

Reply:
Yes, I think that's all right. Removing one of `InstanceInner.id` and `InstanceInner.properties.id` seems like it should be OK to me, and I'm agnostic as to which one we keep.

Reply:
Removed `InstanceInner.id` in d010bab. I kept the other since it's part of the propolis-server API.